### PR TITLE
Remove `html.jelly` in favor of `layout.jelly`

### PR DESCRIPTION
**Pull request by janfaracik**
This has been deprecated since 2017. The only potential uses of it are in private plugins; there are none in the jenkinsci GitHub organisation.

### Testing done

We tracked down the last consumer of `html.jelly` in commit abd7a8f460 by running `git log -- core/src/main/resources/lib/layout/html.jelly`. From there we were able to see that consumers are of the form `<l:html …></l:html>`. From there we did [a code search for `user:jenkinsci "l:html"`](https://cs.github.com/?q=user%3Ajenkinsci+%22l%3Ahtml%22). From there we were able to confirm that there are no more usages of this deprecated API.

### Proposed changelog entries

* The `html.jelly` template has been removed in favor of the more feature- rich `layout.jelly` template.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Review by basil — APPROVED** [review]
The main question we have with regard to PRs like this is whether the PR will break any existing consumers. The PR description is lacking reasoning in this regard. As a courtesy, I have taken the liberty to update the PR description with the following text:

> We tracked down the last consumer of `html.jelly` in commit abd7a8f460 by running `git log -- core/src/main/resources/lib/layout/html.jelly`. From there we were able to see that consumers are of the form `<l:html …></l:html>`. From there we did [a code search for `user:jenkinsci "l:html"`](https://cs.github.com/?q=user%3Ajenkinsci+%22l%3Ahtml%22). From there we were able to confirm that there are no more usages of this deprecated API.

Given the lack of consumers from the code search, I am approving this PR. Please study the reasoning given above and use this as a model for future PR submissions.

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
